File: AudioGeneratorClient/app/src/main/java/com/cesar31/audiogeneratorclient/control/ResponseParserHandler.java
package com.cesar31.audiogeneratorclient.control;

import android.os.Build;

import androidx.annotation.RequiresApi;

import com.cesar31.audiogeneratorclient.model.Err;
import com.cesar31.audiogeneratorclient.model.Playlist;
import com.cesar31.audiogeneratorclient.model.Track;
import com.cesar31.audiogeneratorclient.parser.ClientLex;
import com.cesar31.audiogeneratorclient.parser.ClientParser;

import java.io.StringReader;
import java.util.List;

public class ResponseParserHandler {

    public ResponseParserHandler() {
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public void parseResponse(String input) {
        ClientLex lex = new ClientLex(new StringReader(input));
        ClientParser parser = new ClientParser(lex);
        try {
            parser.parse();
            if(parser.getErrors() != null) {
                // Errores
                List<Err> errors = parser.getErrors();
                errors.forEach(System.out::println);

            } else if(parser.getPlaylists() != null) {
                // listado de listas
                List<Playlist> playlists = parser.getPlaylists();
                playlists.forEach(System.out::println);

            } else if(parser.getTracks() != null) {
                // listado de pistas
                List<Track> tracks = parser.getTracks();
                tracks.forEach(System.out::println);

            } else if(parser.getPlaylist() != null) {
                // playlist especifica con listado de pistas
                Playlist play = parser.getPlaylist();
                System.out.println(play.toString());
                play.getTracks().forEach(System.out::println);

            } else if(parser.getTrack() != null) {
                // pista con sus respectivas notas
                Track track = parser.getTrack();
                System.out.println(track.toString());
                track.getNotes().forEach(System.out::println);

            }

        } catch (Exception e) {
            e.printStackTrace(System.out);
        }
    }
}

File: AudioGeneratorClient/app/src/main/java/com/cesar31/audiogeneratorclient/control/ResponseParserHandler.kt
package com.cesar31.audiogeneratorclient.control

import android.os.Bundle
import com.cesar31.audiogeneratorclient.parser.ClientLex
import com.cesar31.audiogeneratorclient.parser.ClientParser
import java.io.Serializable
import java.io.StringReader
import java.lang.Exception

class ResponseParserHandler {
    fun parseResponse(input: String?): Bundle {
        val bundle = Bundle()
        val lex = ClientLex(StringReader(input))
        val parser = ClientParser(lex)
        try {
            parser.parse()
            when {
                parser.errors != null -> {
                    // Errores
                    val errors = parser.errors
                    bundle.putSerializable("errors", errors as Serializable)
                    return bundle
                }
                parser.playlists != null -> {
                    // listado de listas
                    val playlists = parser.playlists
                    bundle.putSerializable("playlists", playlists as Serializable)
                    return bundle
                }
                parser.tracks != null -> {
                    // listado de pistas
                    val tracks = parser.tracks
                    bundle.putSerializable("tracks", tracks as Serializable)
                    return bundle
                }
                parser.playlist != null -> {
                    // playlist especifica con listado de pistas
                    val play = parser.playlist
                    bundle.putSerializable("play", play)
                    return bundle
                }
                parser.track != null -> {
                    // pista con sus respectivas notas
                    val track = parser.track
                    bundle.putSerializable("track", track)
                    return bundle
                }
            }
        } catch (e: Exception) {
            e.printStackTrace(System.out)
        }
        return bundle
    }
}